### ISSUE TITLE
Error: 'Value cannot be null. Parameter name: propertyInfo.Target' when calling UpdateEntryAsync

### ISSUE THREAD

**Issue by rosspace** [object/Simple.OData.Client#126]
I only get this error when updating but everything works for inserts. It occurs in the constructor for `EdmDeltaModel` in the V4 adapter. There is a foreach that is looping through the navigation properties of the model and for each it builds an `EdmNavigationPropertyInfo` object and passes it to `_entityType.AddUnidirectionalNavigation(navInfo)`. Inside that function call it throws an exception because the Target property of `EdmNavigationPropertyInfo` is null.

You are setting that property with:
```C#
Target = property.Partner != null 
   ? property.Partner.DeclaringEntityType()
   : property.Type.TypeKind() == EdmTypeKind.Collection
   ? (property.Type.Definition as IEdmCollectionType).ElementType.Definition as IEdmEntityType
   : null
```
`property.Partner` is null and it is not a collection so Target gets set to null. I'm not sure what this code is trying to accomplish but is it possible to skip the call to `AddUnidirectionalNavigation` if the Target property is null? Is there something I need to do to get the `property.Partner` field to not be null?

**Comment by object** [comment]
Hi,
Thank you for reporting the issue. Can you send me the metadata file at vagif.abilov@gmail.com? Together with a code example that reproduces the issue.

**Comment by object** [comment]
I believed I reproduced the issue on a different example. Fix is coming soon.

**Comment by object** [comment]
There is fix available at MyGet.org (https://www.myget.org/gallery/simple-odata-client), please give it a try.

**Comment by object** [comment]
Fixed in 4.8.
